### ISSUE TITLE
Integrate /owners routing with tasks

### ISSUE THREAD

**Issue by lyondhill** [influxdata/influxdb#11427]
Proper integration will be a somewhat involved change:

- [x] On task create, assign owning user and org via UserResourceMappingService
- [x] On task delete, clean up UserResourceMappingService
- [x] FindTasks needs to look up task IDs through the UserResourceMappingService
- [x] Task store needs to stop maintaining its own user and owner index
- [ ] Remove task.Owner from the task struct.

**Comment by lyondhill** [comment]
The next steps of this process needs to be put on hold.

In order to update the task find function to use the UserResourceMap to pull a list of tasks by a user id. The current implementation of URM is using a different backend storage than tasks. We need to unify tasks with the other storage systems before we can reasonably make a clean change here

**Comment by mark-rushakoff** [comment]
As discussed in Slack:

To unblock FindTasks integration with users/members, the HTTP task handler should make one call to the URM to discover the valid task IDs, and then make many individual calls to `task/backend.(store.FindTaskByID)`. It certainly won't be performant, but it will be correct. Moving forward on this work will leave us better informed on what we need from the eventual refactor simplifying the task backend.
